### PR TITLE
impr(dev): add endpoint to create test user/data (fehmer)

### PR DESCRIPTION
**Pull request by fehmer**
Allow easy creation of test users and test results on local development environment.

Adds a simple UI only available on dev to create test results and optionally also create test users.
Test users will use  `<username>@example.com` as email and the password is always `password`. You can choose the timeframe in which results are created and  `min/max` amount of results per day. The result count, test type, accuracy, wpm etc will be randomized.

### CODE REVIEW

**Comment by monkeytypegeorge** [comment]
Continuous integration check(s) failed. Please review the failing check's logs and make the necessary changes. https://github.com/monkeytypegame/monkeytype/actions/runs/9110631928

**Comment by monkeytypegeorge** [comment]
Continuous integration check(s) failed. Please review the failing check's logs and make the necessary changes. https://github.com/monkeytypegame/monkeytype/actions/runs/9110677045

File: backend/src/api/routes/index.ts [hunk 1]
```diff
@@ -44,6 +45,7 @@ const API_ROUTE_MAP = {
   "/ape-keys": apeKeys,
   "/admin": admin,
   "/webhooks": webhooks,
+  "/dev": dev,
```
Comment (Miodec) [inline backend/src/api/routes/index.ts:48]: If you do it this way, there is a possibility this gets active in production.

Use the `if (isDevEnvironment())` check below and add it there, then remove it from the configuration.

**Review by Miodec — CHANGES_REQUESTED** [review]

**Comment by monkeytypegeorge** [comment]
Continuous integration check(s) failed. Please review the failing check's logs and make the necessary changes. https://github.com/monkeytypegame/monkeytype/actions/runs/9354195639

**Review by Miodec — APPROVED** [review]
Thanks

**Pull request merged**